hotel_reservation/models/hotel_reservation.py

```diff
@@ -394,19 +432,27 @@ def _create_folio(self):
                                       DEFAULT_SERVER_DATETIME_FORMAT)[:5]))
             for line in reservation.reservation_line:
                 for r in line.reserve:
+                    prod = r.product_id.id
+                    partner = reservation.partner_id.id
+                    folio_line_obj = self.env['hotel.folio.line']
+#                    prod_val = folio_line_obj.product_id_change()
+#                    prod_uom = prod_val['value'].get('product_uom', False)
+#                    price_unit = prod_val['value'].get('price_unit', False)
                     folio_lines.append((0, 0, {
                         'checkin_date': checkin_date,
                         'checkout_date': checkout_date,
                         'product_id': r.product_id and r.product_id.id,
                         'name': reservation['reservation_no'],
-                        'product_uom': r['uom_id'].id,
-                        'price_unit': r['lst_price'],
-                        'product_uom_qty': ((date_a - date_b).days) + 1
-                    }))
+#                        'product_uom': prod_uom,
+                        'product_uom_qty': ((date_a - date_b).days) + 1,
+                        'is_reserved': True}))
                     res_obj = room_obj.browse([r.id])
                     res_obj.write({'status': 'occupied', 'isroom': False})
             folio_vals.update({'room_lines': folio_lines})
             folio = hotel_folio_obj.create(folio_vals)
+            if folio:
+                for rm_line in folio.room_lines:
+                    rm_line.product_id_change()
             self._cr.execute('insert into hotel_folio_reservation_rel'
                              '(order_id, invoice_id) values (%s,%s)',
                              (reservation.id, folio.id)
```

Comment:
Write comments in the method for every little algorithm/logic used.I am sure you can optimise